### PR TITLE
[ILUVATAR_GPU] Add several kernels for PaddleScience examples

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
This pull request introduces several enhancements and new features to the Iluvatar GPU backend, focusing on kernel registration, improved error handling in CMake, and expanded support for sparse and normalization operations. The most significant changes are the addition of new CUDA kernel registrations (including double and triple gradient support), new kernel implementations for instance normalization and sparse transpose operations, and improved CMake checks for missing source files.

**Kernel Registration and Support Expansion:**

* Added registration for new gradient kernels: `sigmoid_double_grad`, `sigmoid_triple_grad`, and `pow_double_grad`, enabling higher-order derivative computations for these activation functions (`activation_grad_kernel_register.cc`). [[1]](diffhunk://#diff-b282ba2674962776f02ea1fc61b5180651ea78aaa7052c9ff745e3b81c247cc3R328-R343) [[2]](diffhunk://#diff-b282ba2674962776f02ea1fc61b5180651ea78aaa7052c9ff745e3b81c247cc3R457-R466)
* Added support for `phi::dtype::float16` in `sigmoid_cross_entropy_with_logits` and its gradient kernel registrations, improving mixed-precision training compatibility (`sigmoid_cross_entropy_with_logits_kernel_register.cu`, `sigmoid_cross_entropy_with_logits_grad_kernel_register.cu`). [[1]](diffhunk://#diff-44424dff6c0e312020083001d03bcb80ee243a8dd398b97b9c505e492aa14854R24) [[2]](diffhunk://#diff-7c3d10fe7c7f1f3e8b24b321b01719fa084a11e2fde2a50ef73d2b4a42a7e267R23)

**Sparse Tensor Operations:**

* Introduced kernel registrations for sparse COO/CSR transpose and their gradients, supporting a wide range of data types including `float16`, `float`, `int`, and `bool` (`sparse_transpose_kernel_register.cu`, `sparse_transpose_grad_kernel_register.cu`). [[1]](diffhunk://#diff-cac075e64f3a18aa5a25f7134b28e23ab4584017ee24b9746fc165a8cb0c09efR1-R43) [[2]](diffhunk://#diff-45db0ead3f92c45c891d4248b52beab7ef2c49847cdca09ad11b4d1963052fafR1-R43)
* Added `values_coo` and `values_coo_grad` kernel registrations for COO sparse tensor value manipulation, and removed unsupported types from `sparse_coo_tensor` and its gradient registration (`sparse_utils_kernel_register.cu`, `sparse_utils_grad_kernel_register.cc`). [[1]](diffhunk://#diff-2b291961e83b0644cf8ff23ceb1f9d4e9fc98cf26a437af46c74e25b1b1d6b25R18-R45) [[2]](diffhunk://#diff-146548b2e6c94a6c391c5bf9e502c32067b19f4f9327b2ba2d922b3731282e94L23-R45)

**Instance Normalization:**

* Implemented a new CUDA kernel for `instance_norm` with full kernel registration, including support for `float`, `float16`, and `bfloat16`, and added utility functions for parameter broadcasting and reduction (`instance_norm_kernel.cu`, `instance_norm_utils.h`). [[1]](diffhunk://#diff-f68595c6cdcb8ca252e56d5d301781f330c0fb682dc6839c20fc2a08e08a17fdR1-R283) [[2]](diffhunk://#diff-221185691b5d9305645f4f1e08d5348de86cedec0db75e2ecce270cc5371995aR1-R75)

**Build System Improvements:**

* Enhanced CMake scripts to check for the existence of all CUDA source files (`CUDA_SRCS1` and `CUDA_SRCS2`), causing a build failure if any file is missing, which improves build reliability (`CMakeLists.txt`). [[1]](diffhunk://#diff-3b1f2093490bd018261f3ef4feff4ab63ee1b3a49378ba70e73268eae268ce17R267-R272) [[2]](diffhunk://#diff-3b1f2093490bd018261f3ef4feff4ab63ee1b3a49378ba70e73268eae268ce17R907-R913)

These changes collectively improve the robustness, feature set, and data type support of the Iluvatar GPU backend, especially for advanced deep learning operations and sparse tensor workflows.

### CODE REVIEW

File: backends/iluvatar_gpu/CMakeLists.txt [hunk 1]
```diff
@@ -264,6 +264,12 @@ file(
   ${PADDLE_SOURCE_DIR}/paddle/phi/kernels/gpu/fill_diagonal_tensor_grad_kernel.cu
   ${PADDLE_SOURCE_DIR}/paddle/phi/kernels/gpu/gelu_grad_kernel.cu)
 
+foreach(src ${CUDA_SRCS1})
```
Comment (YqGe585) [inline backends/iluvatar_gpu/CMakeLists.txt:267]: 这个需要保留吗？

**Review by YqGe585 — APPROVED** [review]
